components/action-mgt/org.wso2.carbon.identity.action.execution/src/test/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImplTest.java

```diff
@@ -0,0 +1,438 @@
+    private ActionExecutionResponseProcessor actionExecutionResponseProcessor;
+    @Mock
+    private APIClient apiClient;
+    @InjectMocks
+    private ActionExecutorServiceImpl actionExecutorService;
+
+    private MockedStatic<ActionExecutionRequestBuilderFactory> actionExecutionRequestBuilderFactory;
+    private MockedStatic<ActionExecutionResponseProcessorFactory> actionExecutionResponseProcessorFactory;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        MockitoAnnotations.openMocks(this);
+        ActionExecutionServiceComponentHolder actionExecutionServiceComponentHolder =
+                ActionExecutionServiceComponentHolder.getInstance();
+        actionExecutionServiceComponentHolder.setActionManagementService(actionManagementService);
+        // Set apiClient field using reflection
+        setField(actionExecutorService, "apiClient", apiClient);
+
+        actionExecutionRequestBuilderFactory = mockStatic(ActionExecutionRequestBuilderFactory.class);
+        actionExecutionResponseProcessorFactory = mockStatic(ActionExecutionResponseProcessorFactory.class);
+    }
+
+    @AfterMethod
+    public void tearDown() {
+
+        actionExecutionRequestBuilderFactory.close();
+        actionExecutionResponseProcessorFactory.close();
+    }
+
+    @Test
+    public void testActionExecuteFailureWhenNoActionsAvailableForActionType() throws Exception {
+
+        when(actionManagementService.getActionsByActionType(any(), any())).thenReturn(Collections.emptyList());
+
+        ActionExecutionStatus actionExecutionStatus =
+                actionExecutorService.execute(ActionType.PRE_ISSUE_ACCESS_TOKEN, any(), any());
+        assertEquals(actionExecutionStatus.getStatus(), ActionExecutionStatus.Status.FAILED);
+    }
+
+    @Test(expectedExceptions = ActionExecutionException.class,
+            expectedExceptionsMessageRegExp = "Multiple actions found for action type: PRE_ISSUE_ACCESS_TOKEN. " +
+                    "Current implementation doesn't support multiple actions for a single action type.")
+    public void testActionExecuteFailureWhenMultipleActionsAvailableForActionType() throws Exception {
+
+        List<Action> mockActions = Arrays.asList(mock(Action.class), mock(Action.class));
+        when(actionManagementService.getActionsByActionType(any(), any())).thenReturn(mockActions);
+
+        actionExecutorService.execute(ActionType.PRE_ISSUE_ACCESS_TOKEN, any(), any());
+    }
+
+    @Test(expectedExceptions = ActionExecutionException.class,
+            expectedExceptionsMessageRegExp = "No request builder found for action type: PRE_ISSUE_ACCESS_TOKEN")
+    public void testActionExecuteFailureWhenNoRegisteredRequestBuilderForActionType() throws Exception {
+
+        when(actionManagementService.getActionsByActionType(any(), any())).thenReturn(
+                Collections.singletonList(mock(Action.class)));
+
+        actionExecutorService.execute(ActionType.PRE_ISSUE_ACCESS_TOKEN, any(), any());
```

Comment:
Shall we add an assert here for actionExecutionStatus. Check other places as well.

Reply:
What's expected here is the exception. So there's really no need to assert the status